Migration to package format 2

fmessmer opened the pull request:
I also fixed the ugly trailing whitespaces with 

```
find -not -path "*.git/*" -type f | xargs sed -i 's/[ \t]*$//'
```

Use [this link](https://github.com/ipa320/cob_common/pull/164?w=1) to suppress whitespace-only changes!

@ipa-fmw @ipa-nhg @ipa-mdl @ipa-mig please check whether I considered everything as discussed

fmessmer committed:
migrate to package format 2

fmessmer committed:
remove trailing whitespaces

fmessmer committed:
remove trailing whitespaces

mgruhler commented:
LGTM

mathias-luedtke commented:
PR is fine, but there a some issue that should be handled in addition:
- Catkin requires CMake version 2.8.3 or higher.
- What is test_urdf.py? If it is a unit test it should not be installed and exec_depends would turn into test_depend.

fmessmer committed:
fix catkin_minimum_required version

fmessmer committed:
add explicit exec_depend to xacro

fmessmer commented:
- fixed catkin_minimum_required
- but I don't know about the test_urdf.py in particular...as it seems to work the way it is, I'd postpone revision of test-related stuff
- I also added an explicit `exec_depend` to `xacro`

fmessmer committed:
remove obsolete autogenerated mainpage.dox files

NadiaHG commented:
LGTM
